packages/twenty-server/src/modules/workflow/workflow-executor/workspace-services/workflow-executor.workspace-service.ts

```diff
@@ -1,17 +1,21 @@
 import { Injectable } from '@nestjs/common';
 
-import { WorkflowStep } from 'src/modules/workflow/workflow-executor/types/workflow-action.type';
-import {
-  WorkflowExecutorException,
-  WorkflowExecutorExceptionCode,
-} from 'src/modules/workflow/workflow-executor/exceptions/workflow-executor.exception';
+import { WorkflowRunStatus } from 'src/modules/workflow/common/standard-objects/workflow-run.workspace-entity';
 import { WorkflowActionFactory } from 'src/modules/workflow/workflow-executor/factories/workflow-action.factory';
+import {
+  WorkflowActionType,
+  WorkflowStep,
+} from 'src/modules/workflow/workflow-executor/types/workflow-action.type';
 
 const MAX_RETRIES_ON_FAILURE = 3;
 
-export type WorkflowExecutionOutput = {
-  result?: object;
-  error?: object;
+export type WorkflowExecutorOutput = {
+  steps: {
+    type: WorkflowActionType;
+    result: object | undefined;
+    error: object | undefined;
+  }[];
+  status: WorkflowRunStatus;
 };
```

Comment:
I think it would be great to store two more things per step:

- The step id; it will be easier to connect the output with the step node on the front-end
- The attemptCount; it would make it possible to store the output of all the attempts. Two attempts can fail for different reasons, and I think storing all of them would be worth it.